instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CoreRerouteTest.kt

```diff
@@ -200,4 +212,75 @@ class CoreRerouteTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.jav
             check(newWaypoints[1] == mockRoute.routeWaypoints.last())
         }
     }
+
+    @Test
+    fun reroute_is_cancelled_when_the_user_returns_to_route() = sdkTest {
+        val responseDelay = 4000L
+        val mockRoute = RoutesProvider.dc_very_short(activity)
+        val originLocation = mockRoute.routeWaypoints.first()
+        val initialLocation = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = originLocation.latitude()
+            longitude = originLocation.longitude()
+        }
+        val offRouteLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = originLocation.latitude() + 0.002
+            longitude = originLocation.longitude()
+        }
+
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+        val requestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = readRawFileText(activity, R.raw.reroute_response_dc_very_short),
+            expectedCoordinates = mockRoute.routeWaypoints,
+            relaxedExpectedCoordinates = false
+        )
+        val rerouteRequestHandler = MockDirectionsRequestHandler(
+            profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+            jsonResponse = readRawFileText(activity, R.raw.reroute_response_dc_very_short),
```

Comment:
nit to add empty (non-valid) directions response
```suggestion
            jsonResponse = readRawFileText(activity, R.raw.empty_directions_response),
```